Core/include/Acts/TrackFinder/CombinatorialKalmanFilter.hpp

```diff
@@ -0,0 +1,1255 @@
+              sourcelinkShared = true;
+            }
+
+            // Add a measurement/outlier track state proxy in multi trajectory
+            // No storage allocation for:
+            // -> predicted parameter and uncalibrated measurement if already
+            // stored
+            // -> filtered parameter for outlier
+            auto stateMask =
+                (predictedShared ? ~TrackStatePropMask::Predicted
+                                 : TrackStatePropMask::All) &
+                (sourcelinkShared ? ~TrackStatePropMask::Uncalibrated
+                                  : TrackStatePropMask::All) &
+                (isOutlier ? ~TrackStatePropMask::Filtered
+                           : TrackStatePropMask::All);
+            auto trackTip =
+                result.fittedStates.addTrackState(stateMask, result.currentTip);
+
+            // Get the track state proxy
+            auto trackStateProxy = result.fittedStates.getTrackState(trackTip);
+
+            // Fill the track state proxy
+            if (predictedShared) {
+              // The predicted parameter is already stored, just set the index
+              auto sharedPredicted =
+                  result.fittedStates.getTrackState(neighborTip);
+              trackStateProxy.data().ipredicted =
+                  sharedPredicted.data().ipredicted;
+            } else {
+              trackStateProxy.predicted() = boundParams.parameters();
+              trackStateProxy.predictedCovariance() = *boundParams.covariance();
+            }
+            trackStateProxy.jacobian() = jacobian;
+            trackStateProxy.pathLength() = pathLength;
+
+            // Get and set the type flags
+            auto& typeFlags = trackStateProxy.typeFlags();
+            typeFlags.set(TrackStateFlag::MaterialFlag);
+            typeFlags.set(TrackStateFlag::ParameterFlag);
+            if (isOutlier) {
+              // Set the outlier type flag
+              typeFlags.set(TrackStateFlag::OutlierFlag);
+            } else {
+              // Set the measurement type flag
+              typeFlags.set(TrackStateFlag::MeasurementFlag);
+            }
+
+            // Assign the source link and calibrated measurement to the track
+            // state
+            if (sourcelinkShared) {
+              // The source link is already stored, just set the index
+              auto sharedMeasurement =
+                  result.fittedStates.getTrackState(index_it->second);
+              trackStateProxy.data().iuncalibrated =
+                  sharedMeasurement.data().iuncalibrated;
+            } else {
+              trackStateProxy.uncalibrated() = sourcelinks.at(index);
+            }
+            std::visit(
+                [&](const auto& calibrated) {
+                  trackStateProxy.setCalibrated(calibrated);
+                },
+                m_calibrator(trackStateProxy.uncalibrated(),
+                             trackStateProxy.predicted()));
```

Comment:
Maybe I'm confused, but could the calibrated measurement not be shared as well? Neighbor can have the same predicted parameters, you check that above, right? In that case, calibrated should also be identical, no?

Reply:
This is indeed confusing. We have no way to know if the calibrated measurement is shared or not since the source links are shared by different tracks, i.e. the predicted parameters could be different for those shared source links. Does this sound reasonable?

Reply:
But don't you have `predictedShared` just a few lines above from here? And if `predictedShared && sourceLinkShared`, shouldn't `calibrated` be identical then?

Reply:
The `predictedShared` and `sourcelinkShared` is just a boolean. It could be `false`, then not shared...

Reply:
I think I need to rename them to  `isPredictedShared` and `isSourcelinkShared`.